src/main/java/com/example/demo/Controller/StudentController.java

```diff
@@ -1,27 +1,75 @@
 package com.example.demo.Controller;
+
 import com.example.demo.Service.StudentService;
 import com.example.demo.StudentModel.Student;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import io.swagger.v3.oas.annotations.tags.Tags;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.server.DelegatingServerHttpResponse;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.PathVariable;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
-
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.client.HttpClientErrorException;
 
 import java.util.List;
+import java.util.Optional;
 
 
 @RestController
 @RequestMapping("/students")
 public class StudentController {
 
-      @Autowired
+    @Autowired
     private StudentService studentService;
 
-      @GetMapping
-    public List<Student> getAllStudents() {
+    @GetMapping
+    @Tag(name = "List")
+    public ResponseEntity<List<Student>> getAllStudents() {
+
+        List<Student> students = studentService.getAllStudents();
+        return ResponseEntity.ok(students);
+    }
+
+    @GetMapping("/{id}")
+    @Tag(name = "List")
+    public ResponseEntity<Student> getStudentById(@PathVariable Long id) {
+        Optional<Student> studentOptional = studentService.getStudentById(id);
+        return studentOptional
+                .map(student -> ResponseEntity.ok(student))
+                .orElseGet(() -> ResponseEntity.status(HttpStatus.NOT_FOUND).build());
+    }
+
+    @PostMapping
+    @Tag(name = "Add")
+    public ResponseEntity<Void> addStudent(@RequestBody Student student) {
+        studentService.addStudent(student);
+        return ResponseEntity.status(HttpStatus.CREATED).build();
```

Comment:
Same, we need to return the student that we just created.

Right now you are passing the student from outside and using the same models, but later we will specialise them with **StudentRequestDTO** and **StudentResponseDTO**